From 45ef6ece3bb8fb52a9fed6269ee9e44d8f774107 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:04:37 -0400
Subject: [fs] btrfs: prevent RCU warning when dereferencing radix tree slot

Message-id: <1407179109-13714-391-git-send-email-sandeen@redhat.com>
Patchwork-id: 87267
O-Subject: [RHEL7.1 PATCH 390/422] btrfs: prevent RCU warning when dereferencing radix tree slot
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Sasha Levin <sasha.levin@oracle.com>

commit f1e3c289498f26d98b65ddaae0cd892c98a6ce84
Author: Sasha Levin <sasha.levin@oracle.com>
Date:   Wed Jun 11 12:00:25 2014 -0400

    btrfs: prevent RCU warning when dereferencing radix tree slot

    Mark the dereference as protected by lock. Not doing so triggers
    an RCU warning since the radix tree assumed that RCU is in use.

    Signed-off-by: Sasha Levin <sasha.levin@oracle.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/tests/btrfs-tests.c b/fs/btrfs/tests/btrfs-tests.c
index a5dcacb..9626252 100644
--- a/fs/btrfs/tests/btrfs-tests.c
+++ b/fs/btrfs/tests/btrfs-tests.c
@@ -135,7 +135,7 @@ restart:
  radix_tree_for_each_slot(slot, &fs_info->buffer_radix, &iter, 0) {
   struct extent_buffer *eb;
 
-  eb = radix_tree_deref_slot(slot);
+  eb = radix_tree_deref_slot_protected(slot, &fs_info->buffer_lock);
   if (!eb)
    continue;
   /* Shouldn't happen but that kind of thinking creates CVE's */
-- 
1.7.1